lib/sshutils/server.go

```diff
@@ -125,15 +127,25 @@ func (s *Server) Wait() {
 
 // Close closes listening socket and stops accepting connections
 func (s *Server) Close() error {
-	return s.listener.Close()
+	s.askedToClose = true
+	if s.listener != nil {
+		return s.listener.Close()
+	}
+	return nil
 }
 
 func (s *Server) acceptConnections() {
 	defer s.notifyClosed()
-	log.Infof("%v ready to accept connections", s.Addr())
+	addr := s.Addr()
+	log.Infof("%v ready to accept connections", addr)
 	for {
 		conn, err := s.listener.Accept()
 		if err != nil {
+			if s.askedToClose {
```

Comment:
same here - looks like data race, I would try to use broadcast closer `IsClose` method here to protect from this potential race